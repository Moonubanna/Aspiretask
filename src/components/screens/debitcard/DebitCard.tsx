/* eslint-disable react-native/no-inline-styles */
import React, {useState, useContext, useEffect} from 'react';
import {View} from 'react-native';
import {useSelector} from 'react-redux';
import {useDispatch} from 'react-redux';
import {shallowEqual} from 'react-redux';
import {SCREEN} from '../../../constants';
import {
  ACCOUNT,
  WEEKLY,
  FREEZE,
  NEW_CARD,
  DEACTIVATED,
} from '../../../common/images';
import translate from '../../../i18n/i18n';
import {BookContext} from '../../../contexts';
import * as Utils from '../../../utils';
import * as Handlers from './handlers';
//Hoc
import * as HOC from '../../../common/hoc';
import styles from './styles';
import headerComponent from './child_components/HeaderComponent';
import cardComponent from './child_components/CardComponent';

import {
  getDebitCardRequestDataAction,
  getClearDebitCardRequestDataAction,
} from '../../../actions';

const HeaderHOCView = HOC.HeaderHOC(View);

const DebitCard = ({}) => {
  const {navigation, route} = useContext(BookContext);
  const dispatch = useDispatch();
  const {debitCardResponse, loading} = useSelector(
    state => ({
      debitCardResponse: state.getDebitCardRequestCasesReducer.response,
      loading: state.getDebitCardRequestCasesReducer.loading,
    }),
    shallowEqual,
  );

  const [savedUserData, setSavedUserData] = useState(undefined);
  const [availableBalance, setAvailableBalance] = useState('5000');
  const [isShowCard, setIsShowCard] = useState(false);
  const [cardReponse, setCardResponse] = useState(undefined);

  const [cardChildArray, setCardChildArray] = useState([
    {
      id: 1,
      name: translate('TOP_UP_ACCOUNT'),
      info: translate('TOP_UP_ACCOUNT_INFO'),
      image: ACCOUNT,
      is_radio: false,
      is_radio_enable: false,
    },
    {
      id: 2,
      name: translate('WEEKLY_SPENDING_LIMIT'),
      info: translate('WEEKLY_SPENDING_LIMIT_INFO'),
      image: WEEKLY,
      is_radio: true,
      is_radio_enable: false,
    },
    {
      id: 3,
      name: translate('FREEZE_CARD'),
      info: translate('FREEZE_CARD_INFO'),
      image: FREEZE,
      is_radio: true,
      is_radio_enable: false,
    },
    {
      id: 4,
      name: translate('GET_A_NEW_CARD'),
      info: translate('GET_A_NEW_CARD_INFO'),
      image: NEW_CARD,
      is_radio: false,
      is_radio_enable: false,
    },
    {
      id: 5,
      name: translate('DEACTIVATED_CARDS'),
      info: translate('DEACTIVATED_CARDS_INFO'),
      image: DEACTIVATED,
      is_radio: false,
      is_radio_enable: false,
    },
  ]);

  // Component did mount
  useEffect(() => {
    // initial loader calling before calling comment api
    getDebitCardApi();
    // component will unmount
    return () => {
      dispatch(getClearDebitCardRequestDataAction(undefined));
    };
  }, []);

  /* debit card response handling */
  useEffect(() => {
    if (debitCardResponse) {
      //Handler function for handle post save comment response
      const handleResponse = Handlers.setDebitCardData(debitCardResponse);
      if (handleResponse) {
        setCardResponse(handleResponse.data);
      }
    }
  }, [debitCardResponse]);

  // Call get debit card api
  function getDebitCardApi() {
    let requestObj = {
      api_type: 'GET',
    };
    Handlers.callGetDebitCardApi(
      requestObj,
      dispatch,
      getDebitCardRequestDataAction,
    );
  }
  function pressBack() {
    navigation.goBack();
  }

  function pressRightIcon() {}
  function pressCardVisibility() {
    setIsShowCard(!isShowCard);
  }
  function pressRadio(item: object, index: number) {
    if (item.id === 2) {
      // Navigate to weekly limit screen
      Utils.navigateWithParams(navigation, SCREEN.WEEKLYLIMIT, {
        data: {object: item, index: index},
        updateWeeklyLimit,
      });
    }
  }
  function updateWeeklyLimit(item: object, index: number, amount: string) {
    let savedCardChildArray = cardChildArray;
    savedCardChildArray[index].is_radio_enable = true;
    savedCardChildArray[index].info = `${translate(
      'WEEKLY_SPENDING_LIMIT_SET_INFO',
    )} ${amount}`;

    setCardChildArray([...savedCardChildArray]);
  }
  return (
    <HeaderHOCView
      isLoading={loading}
      backPress={() => {
        pressBack();
      }}
      onPressIcon={() => {
        pressRightIcon();
      }}
      style={{flex: 1}}
      isHeader={false}
      isRightIcon={false}>
      <View style={styles.container}>{headerComponent(availableBalance)}</View>
      {cardComponent(
        isShowCard,
        pressCardVisibility,
        cardChildArray,
        cardReponse,
        pressRadio,
      )}
    </HeaderHOCView>
  );
};

export default DebitCard;
