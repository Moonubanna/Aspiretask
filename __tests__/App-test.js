/**
 * @format
 */

import 'react-native';
import React from 'react';
import {Provider} from 'react-redux';
import Store from './../src/store';
import DebitCard from './../src/components/screens/debitcard';

// Note: test renderer must be required after react-native.
import {create, act} from 'react-test-renderer';
const store = Store();
const tree = create(
  <Provider store={store}>
    <DebitCard />
  </Provider>,
);

test('snapshot', () => {
  expect(tree).toMatchSnapshot();
});
