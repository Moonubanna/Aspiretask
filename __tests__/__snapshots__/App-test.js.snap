// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#0C365A",
          "flex": 1,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "background": "#43366E",
            "flexDirection": "column",
            "padding": 15,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "column",
              "paddingTop": 20,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#ffffff",
                "fontFamily": "AirbnbCerealApp-Bold",
                "fontSize": 20,
              }
            }
          >
            Debit Card
          </Text>
          <Text
            style={
              Object {
                "color": "#ffffff",
                "fontFamily": "AirbnbCerealApp-Book",
                "fontSize": 14,
                "marginTop": 20,
              }
            }
          >
            Available Balance
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "marginTop": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#01D167",
                  "borderRadius": 5,
                  "paddingHorizontal": 14,
                  "paddingVertical": 3,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "AirbnbCerealApp-Medium",
                    "fontSize": 14,
                  }
                }
              >
                S$
              </Text>
            </View>
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontFamily": "AirbnbCerealApp-Bold",
                  "fontSize": 20,
                  "marginLeft": 7,
                }
              }
            >
              5000
            </Text>
          </View>
        </View>
        <Image
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../src/assets/images/logo.png",
            }
          }
          style={
            Object {
              "height": 25,
              "position": "absolute",
              "right": 15,
              "top": 15,
              "width": 25,
            }
          }
        />
      </View>
    </View>
    <RCTScrollView
      showsVerticalScrollIndicator={false}
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View>
        <View>
          <View
            style={
              Object {
                "marginTop": 200,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#ffffff",
                  "borderTopLeftRadius": 16,
                  "borderTopRightRadius": 16,
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 110,
                }
              }
            />
            <View
              style={
                Object {
                  "backgroundColor": "#00000000",
                  "height": 240,
                  "justifyContent": "flex-end",
                  "margin": 16,
                }
              }
            >
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": "#ffffff",
                      "borderRadius": 5,
                      "flexDirection": "row",
                      "height": 60,
                      "justifyContent": "center",
                      "paddingTop": 8.333333333333334,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "width": 170,
                    },
                    Object {
                      "opacity": 1,
                    },
                  ]
                }
              >
                <Image
                  resizeMode="contain"
                  source={
                    Object {
                      "testUri": "../../../src/assets/images/green_eye_visible.png",
                    }
                  }
                  style={
                    Object {
                      "height": 18,
                      "width": 18,
                    }
                  }
                />
                <Text
                  style={
                    Object {
                      "color": "#01D167",
                      "fontFamily": "AirbnbCerealApp-Medium",
                      "fontSize": 13,
                      "marginLeft": 5,
                    }
                  }
                >
                  Show card number
                </Text>
              </View>
              <View
                style={
                  Object {
                    "backgroundColor": "#01D167",
                    "borderRadius": 16,
                    "height": 210,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flexDirection": "column",
                      "height": 210,
                      "justifyContent": "center",
                      "paddingLeft": 20,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#ffffff",
                        "fontFamily": "AirbnbCerealApp-Bold",
                        "fontSize": 20,
                      }
                    }
                  />
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "marginTop": 10,
                      }
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#ffffff",
                            "fontFamily": "AirbnbCerealApp-Medium",
                            "fontSize": 16,
                            "marginTop": 20,
                          },
                          Object {
                            "textAlignVertical": "center",
                          },
                        ]
                      }
                    >
                      ####  ####  ####
                    </Text>
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#ffffff",
                            "fontFamily": "AirbnbCerealApp-Medium",
                            "fontSize": 16,
                            "marginTop": 20,
                          },
                          Object {
                            "marginLeft": 20,
                          },
                        ]
                      }
                    />
                  </View>
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "marginTop": 10,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "#ffffff",
                          "fontFamily": "AirbnbCerealApp-Medium",
                          "fontSize": 16,
                        }
                      }
                    >
                      
                    </Text>
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#ffffff",
                            "fontFamily": "AirbnbCerealApp-Medium",
                            "fontSize": 16,
                          },
                          Object {
                            "marginLeft": 20,
                          },
                        ]
                      }
                    >
                      
                    </Text>
                  </View>
                </View>
                <Image
                  resizeMode="contain"
                  source={
                    Object {
                      "testUri": "../../../src/assets/images/aspire_hori_logo.png",
                    }
                  }
                  style={
                    Object {
                      "position": "absolute",
                      "right": 20,
                      "top": 20,
                      "width": 70,
                    }
                  }
                />
                <Image
                  resizeMode="contain"
                  source={
                    Object {
                      "testUri": "../../../src/assets/images/visa_logo.png",
                    }
                  }
                  style={
                    Object {
                      "bottom": 20,
                      "position": "absolute",
                      "right": 20,
                      "width": 70,
                    }
                  }
                />
              </View>
            </View>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#ffffff",
                  "flexDirection": "row",
                  "padding": 10,
                  "width": "100%",
                },
                Object {
                  "opacity": 1,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "width": "100%",
                }
              }
            >
              <Image
                resizeMode="contain"
                source={
                  Object {
                    "testUri": "../../../src/assets/images/account.png",
                  }
                }
                style={
                  Object {
                    "height": 35,
                    "width": 35,
                  }
                }
              />
              <View
                style={
                  Object {
                    "flexDirection": "column",
                    "justifyContent": "center",
                    "marginLeft": 10,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#1A1A1A",
                      "fontFamily": "AirbnbCerealApp-Medium",
                      "fontSize": 14,
                    }
                  }
                >
                  Top-up account
                </Text>
                <Text
                  style={
                    Object {
                      "color": "#bdbdbd",
                      "fontFamily": "AirbnbCerealApp-Book",
                      "fontSize": 11,
                      "marginTop": 5,
                    }
                  }
                >
                  Deposit money to your account to use with card
                </Text>
              </View>
            </View>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#ffffff",
                  "flexDirection": "row",
                  "padding": 10,
                  "width": "100%",
                },
                Object {
                  "opacity": 1,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "width": "80%",
                }
              }
            >
              <Image
                resizeMode="contain"
                source={
                  Object {
                    "testUri": "../../../src/assets/images/weekly.png",
                  }
                }
                style={
                  Object {
                    "height": 35,
                    "width": 35,
                  }
                }
              />
              <View
                style={
                  Object {
                    "flexDirection": "column",
                    "justifyContent": "center",
                    "marginLeft": 10,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#1A1A1A",
                      "fontFamily": "AirbnbCerealApp-Medium",
                      "fontSize": 14,
                    }
                  }
                >
                  Weekly spending limit
                </Text>
                <Text
                  style={
                    Object {
                      "color": "#bdbdbd",
                      "fontFamily": "AirbnbCerealApp-Book",
                      "fontSize": 11,
                      "marginTop": 5,
                    }
                  }
                >
                  You haven'n set any spending limit on card
                </Text>
              </View>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "width": "20%",
                }
              }
            >
              <Image
                resizeMode="contain"
                source={
                  Object {
                    "testUri": "../../../src/assets/images/toggle.png",
                  }
                }
                style={
                  Object {
                    "height": 28,
                    "tintColor": "#bdbdbd",
                    "width": 32,
                  }
                }
              />
            </View>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#ffffff",
                  "flexDirection": "row",
                  "padding": 10,
                  "width": "100%",
                },
                Object {
                  "opacity": 1,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "width": "80%",
                }
              }
            >
              <Image
                resizeMode="contain"
                source={
                  Object {
                    "testUri": "../../../src/assets/images/freeze.png",
                  }
                }
                style={
                  Object {
                    "height": 35,
                    "width": 35,
                  }
                }
              />
              <View
                style={
                  Object {
                    "flexDirection": "column",
                    "justifyContent": "center",
                    "marginLeft": 10,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#1A1A1A",
                      "fontFamily": "AirbnbCerealApp-Medium",
                      "fontSize": 14,
                    }
                  }
                >
                  Freeze card
                </Text>
                <Text
                  style={
                    Object {
                      "color": "#bdbdbd",
                      "fontFamily": "AirbnbCerealApp-Book",
                      "fontSize": 11,
                      "marginTop": 5,
                    }
                  }
                >
                  Your debit card is currently active
                </Text>
              </View>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "width": "20%",
                }
              }
            >
              <Image
                resizeMode="contain"
                source={
                  Object {
                    "testUri": "../../../src/assets/images/toggle.png",
                  }
                }
                style={
                  Object {
                    "height": 28,
                    "tintColor": "#bdbdbd",
                    "width": 32,
                  }
                }
              />
            </View>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#ffffff",
                  "flexDirection": "row",
                  "padding": 10,
                  "width": "100%",
                },
                Object {
                  "opacity": 1,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "width": "100%",
                }
              }
            >
              <Image
                resizeMode="contain"
                source={
                  Object {
                    "testUri": "../../../src/assets/images/new_card.png",
                  }
                }
                style={
                  Object {
                    "height": 35,
                    "width": 35,
                  }
                }
              />
              <View
                style={
                  Object {
                    "flexDirection": "column",
                    "justifyContent": "center",
                    "marginLeft": 10,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#1A1A1A",
                      "fontFamily": "AirbnbCerealApp-Medium",
                      "fontSize": 14,
                    }
                  }
                >
                  Get a new card
                </Text>
                <Text
                  style={
                    Object {
                      "color": "#bdbdbd",
                      "fontFamily": "AirbnbCerealApp-Book",
                      "fontSize": 11,
                      "marginTop": 5,
                    }
                  }
                >
                  This deactivates your current debit card
                </Text>
              </View>
            </View>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#ffffff",
                  "flexDirection": "row",
                  "padding": 10,
                  "width": "100%",
                },
                Object {
                  "opacity": 1,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "width": "100%",
                }
              }
            >
              <Image
                resizeMode="contain"
                source={
                  Object {
                    "testUri": "../../../src/assets/images/deactivated.png",
                  }
                }
                style={
                  Object {
                    "height": 35,
                    "width": 35,
                  }
                }
              />
              <View
                style={
                  Object {
                    "flexDirection": "column",
                    "justifyContent": "center",
                    "marginLeft": 10,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#1A1A1A",
                      "fontFamily": "AirbnbCerealApp-Medium",
                      "fontSize": 14,
                    }
                  }
                >
                  Deactivated cards
                </Text>
                <Text
                  style={
                    Object {
                      "color": "#bdbdbd",
                      "fontFamily": "AirbnbCerealApp-Book",
                      "fontSize": 11,
                      "marginTop": 5,
                    }
                  }
                >
                  Your previously deactivated cards
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
